PR_REVIEW_CHECKLIST.md

```diff
@@ -0,0 +1,98 @@
+    - Feel free to ask for help on how to test changes on all platforms.
+2. Make sure the “Fixed Issues” section is formatted correctly. Links should be clickable and lead to the correct issue being fixed by this PR.
+3. Make sure the testing steps are clear and correctly test the changes made. Make note of what’s required for the test account (i.e. do you need to test using a brand new account, have a workspace, have a bank account, etc).
+    - > **Rule of thumb**: Over-explain what the tester needs to do. Do not rely on the tester having existing knowledge of the app.
+      > 
+      > ### Example of bad steps:
+      >
+	  >	1. Go to this page
+	  >	2. Click the button
+      >
+      > ### Example of good steps:
+      >
+      > 1. Log in with an account that has a workspace, or follow `[these steps](link-to-instructions-for setting-up-workspace)` to make one.
+      > 2. Click the settings button on the top left (the one with your avatar)
+      > 3. Navigate to “Your Workspace” > Reimburse Receipts > Connect bank account
+      > 4. Verify that the bank account view opens without errors
+      >
+    - Take note of the distinction between testing _locally_ and _on staging_. Note: The staging site references the production version of the API.
+4. Make sure the testing steps also cover possible regressions.
+    1. Cases where we definitely need new regression tests:
+        1. Building a new page
+        2. Building a new feature
+        3. Building a new component?
+    2. Cases where we might need new regression tests:
+        1. Bug was easy to find
+5. Make sure that the test includes both **success** and **fail** scenarios.
+    - Ex: if the issue was to prevent the user from saving a field when it is empty, besides testing that we should also test that they can save the field _if it is not empty_.
+
+# Testing the changes
+
+1. Make sure there are no console errors related to the PR. If you find console errors in a branch you’re reviewing, check if they also exist in `main`.
+    1. If they do, proceed as normal (report them in #expensify-open-source if you think they need to be fixed).
+    2. If the errors do not exist in `main`, report it to the PR author - they need to be fixed before the PR can be merged.
+2. Test the changes on **all platforms**. Follow this guide (needs link) to set up each platform.
+    - If you’re unable to boot a platform for any reason, ask for help in the #expensify-open-source Slack channel. Your issue might already have been addressed there, so be sure to search first.
+3. Should we add regression testing for code introduced in this PR?
+    - If the PR introduces **new** pages, features, etc. then we **should** have new regression tests written. Please note this in your PR so that Applause can make sure it gets added.
+    - If the PR fixes a bug, Applause will check to make sure the existing regression testing steps already cover the workflow fixed by the bug.
```

Comment:
This possibly seems like something we should just ask Applause to do. It seems like a weird instruction to ask someone to:

- Suggest that a test be added as a regression
- Applause checks if it exists 
- Applause adds it as a regression if not

Why is Applause not identifying the opportunity to just add the regression test? Is this defined in the scope of the work they are contracted to do for us, or not? If not, and we really need contributors to suggest areas to test for regressions we should provide examples of WHAT that task looks like and WHO is responsible to do this.

>Please note this in your PR so that Applause can make sure it gets added.

What does this mean? How does someone "note" this and who are we talking to?

Reply:
> Is this defined in the scope of the work they are contracted to do for us, or not?

This seems like the main question - I agree if they can / are willing to do it themselves, that would definitely be best. Then the other questions don't really need to be answered 😅

Reply:
Asking in #applause [here](https://expensify.slack.com/archives/CAMPKJQ94/p1644235856338909)

Reply:
Hmm @tgolen made a good point in Slack ([here](https://expensify.slack.com/archives/CAMPKJQ94/p1644251179988849?thread_ts=1644235856.338909&cid=CAMPKJQ94)):

> We are the experts on knowing what's valuable to test and what isn't. If we just gave them open permission to create tests, it could result in a bunch of bloat that just increases the price of our contract. It would be in their best interest to create as many tests as they could so they could get paid more.